src/components/title/title.jsx

```diff
@@ -0,0 +1,93 @@
+import { cn } from '../../utilities/functions';
+
+/**
+ * Title component.
+ */
+
+const Title = ( {
+	title = null,
+	description = null,
+	icon = null,
+	iconPosition = 'right', // left, right
+	tag = 'h2', // h1, h2, h3, h4, h5, h6
+	size = 'sm', // xs, sm, md, lg
+	className = '',
+} ) => {
+	// Base classes. - Mandatory classes.
+	const iconGap = {
+		xs: 'flex items-center gap-1',
+		sm: 'flex items-center gap-1',
+		md: 'flex items-center gap-1.5',
+		lg: 'flex items-center gap-1.5',
+	};
+
+	if ( ! title ) {
+		return null;
+	}
+
+	const getTitle = () => {
+		const Tag = tag;
+		const titleCommonClasses = 'font-semibold';
+		// Size classes - Based on the size prop.
+		const sizeClasses = {
+			xs: 'text-base [&>*]:text-base [&>svg]:h-3.5 [&>svg]:w-3.5 gap-1 p-0 m-0',
```

Comment:
This is just a suggestion when the width and height are both the same instead of using `w-x` and `h-x` we can use the `size-x` class name.


```suggestion
			xs: 'text-base [&>*]:text-base [&>svg]:size-3.5 gap-1 p-0 m-0',
```

Reply:
Done